xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -371,15 +373,32 @@ def parse_optional_variable(
                 # Always qualify builtin attributes
                 # This is technically an approximation, but appears to be good enough
                 # for xDSL right now.
-                if unique_base is not None and unique_base in Builtin.attributes:
+                unique_type = None
+                if unique_base is not None and issubclass(unique_base, TypedAttribute):
+                    constr = attr_def.constr
+                    if isinstance(constr, ParamAttrConstraint):
+                        type_constraint = constr.param_constrs[unique_base.type_index]
+                        if type_constraint.can_infer(set()):
+                            unique_type = type_constraint.infer({})
```

Comment:
Note, we should improve this to allow things like `AllOf(ParamAttrConstraint(...), ...)` as well?

Reply:
By improve, I mean adding a comment and an issue for now :)
If you can, also adding the issue number in the comment as well

Reply:
I don't get it
Do you mean improve the .infer to have this working in more cases?